### DISCUSSION TITLE
Intermediate representation of code

### DISCUSSION THREAD

**Discussion by lucas-mior** [OpenModelica/OpenModelica#12376] [Q&A]
Hello, I know that I can use the transformational debugger to check which equations result from the Modelica code.
However, I find that it is confusing. Is there a way to check how OM transforms code before translating into C code?

**Comment by casella** [comment]
@lucas-mior can you elaborate more on "confusing", maybe with some specific example?

If you set the [-d=optdaedump](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/omchelptext.html#omcflag-debug-optdaedump) compiler flag, you get dumps from all modules of the backend. A bit verbose, but the information is there.

**Comment by lucas-mior** [comment]
In the following equation, I would like to know what is the difference between `$DER.variable` and `der(variable)`.
```
(assign) Q := rs * hs * $DER.Vst + (rs * $DER.hs + $DER.rs * hs) * Vst + rw * hw * der(Vwt) + (rw * $DER.hw + $DER.rw * hw) * Vwt + 8415000.0 * $DER.t + (-13.49708445369991) * derpdt.y - (qf * hf - qs * hs)
```

I also would like to understand what do `(torn)` and `(residual)` mean. Maybe that's all I need to know for what I am trying to do here :).
```
(residual) (-13.49708445369991) * derpdt.y + 8415000.0 * $DER.t + (rw * $DER.hw + $DER.rw * hw) * Vwt + rw * hw * $DER.Vwt + (rs * $DER.hs + $DER.rs * hs) * Vst + rs * hs * $DER.Vst + qs * hs - Qinput - qf * hf = 0
(torn) qr := qdc - 1.9 * (rs * $DER.av + $DER.rs * av + $DER.rw * (1.0 - av) - rw * $DER.av)
```

**Reply by casella** [reply]
$DER are dummy derivatives, coming from index reduction algorithms. 

Torn and residual refer to systems solved with the tearing algorithm.

You can check the excellent book by F. Cellier and E. Kofman, Continuous system simulation, to learn more.

**Reply by lucas-mior** [reply]
This book is great. Thanks for the recommendation.
